pkg/controller/useraccount/useraccount_controller_test.go

```diff
@@ -776,6 +776,118 @@ func TestUpdateStatus(t *testing.T) {
 	})
 }
 
+func TestDisabledUserAccount(t *testing.T) {
+	logf.SetLogger(logf.ZapLogger(true))
+	username := "johndoe"
+	userID := uuid.NewV4().String()
+	s := scheme.Scheme
+	err := apis.AddToScheme(s)
+	require.NoError(t, err)
+
+	t.Run("disabled useraccount", func(t *testing.T) {
```

Comment:
This test currently looks like an e2e test. The entire flow will be covered by the e2e tests.
In unit test we need to pre-configure everything then do just one reconcile and check the result.

So it would be something like this:

Test 1. Testing initiating disabling.
- Pre-configuration: create User Account; create User; create Identity, create NSTemplateSet. See https://github.com/codeready-toolchain/member-operator/blob/master/pkg/controller/useraccount/useraccount_controller_test.go#L347 for example
- Set UserAccount.Disabled = true
- Reconcile
- Check: User&Identity is gone. NSTemplateSet is still there. Status is set to "disabling".

Test 2. Testing "disabled" status.
- Pre-configuration: create User Account (disabled == true); create NSTemplateSet. No User or Identity.
- Reconcile
- Check: User&Identity is still missing. NSTemplateSet is still there. Status is set to "disabled".

Tests 3&4. Testing initiating disabling when only User or Identity exist. Almost the same as 1 but covering the use cases when there is only User or Identity exist (not both of them).

Test 5. Testing deletion of the disabled user account.
- pre-configuration is almost the same as in Test 2 but add finalizer to UserAccount + set UserAccount deletion timestamp to something (== being deleted). It imitates deletion of the disabled user account.
- Reconcile
- Check that User&Identity&NSTemplateSet do not exist.

Reply:
fixed!